nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ControllerResource.java

```diff
@@ -2411,6 +2429,289 @@ public Date getEndDate() {
+                    @ApiResponse(responseCode = "401", description = "Client could not be authenticated."),
+                    @ApiResponse(responseCode = "403", description = "Client is not authorized to make this request."),
+                    @ApiResponse(responseCode = "409", description = "The request was valid but NiFi was not in the appropriate state to process it.")
+            }
+    )
+    public Response getNarSummary(
+            @PathParam("id")
+            @Parameter(description = "The id of the NAR.", required = true)
+            final String id) {
+        authorizeController(RequestAction.READ);
+
+        if (StringUtils.isBlank(id)) {
+            throw new IllegalArgumentException("Id is required");
+        }
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.GET);
+        }
+
+        final NarSummaryEntity summaryEntity = serviceFacade.getNarSummary(id);
+        return generateOkResponse(summaryEntity).build();
+    }
+
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/nar-manager/nars/{id}/details")
+    @Operation(
+            summary = "Retrieves the component types available from the installed NARs",
+            responses = @ApiResponse(content = @Content(schema = @Schema(implementation = NarDetailsEntity.class))),
+            security = {
+                    @SecurityRequirement(name = "Read - /controller")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(responseCode = "400", description = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+                    @ApiResponse(responseCode = "401", description = "Client could not be authenticated."),
+                    @ApiResponse(responseCode = "403", description = "Client is not authorized to make this request."),
+                    @ApiResponse(responseCode = "409", description = "The request was valid but NiFi was not in the appropriate state to process it.")
+            }
+    )
+    public Response getNarDetails(
+            @PathParam("id")
+            @Parameter(description = "The id of the NAR.", required = true)
+            final String id) {
+        authorizeController(RequestAction.READ);
+
+        if (StringUtils.isBlank(id)) {
+            throw new IllegalArgumentException("Id is required");
+        }
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.GET);
+        }
+
+        final NarDetailsEntity detailsEntity = serviceFacade.getNarDetails(id);
+        return generateOkResponse(detailsEntity).build();
+    }
+
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Path("/nar-manager/nars/{id}/download")
```

Comment:
We already have a `GET` for `nar-manager/nars/{id}` which returns the summary for the NAR with the given id.

I guess then can be differentiated by the requested content type? 

We also have similar examples like downloading a flow definition vs retrieving JSON for a process process group:
```
/process-groups/{id}/download
/process-groups/{id}
```

Reply:
Decided to use `content` on the upload and download end-points, so it will be:
```
POST /nar-manager/nars/content
GET /nar-manager/nars/{id}/content
```